### PR TITLE
Rename Multi plugin to ov_multi_plugin.

### PR DESCRIPTION
**Pull request by yangwang201911**
Signed-off-by: Wang, Yang <yang4.wang@intel.com>

### Details:
 - MultiDevicePlugin-> ov_multi_plugin
 - Move the related src files to new place

### Tickets:
 - https://jira.devtools.intel.com/browse/CVS-71048

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
I suppose we also need to move plugin to a new place and rename local files like in PR https://github.com/openvinotoolkit/openvino/pull/8872/

**Comment by yangwang201911** [comment]
> I suppose we also need to move plugin to a new place and rename local files like in PR #8872
Yes, we will move those files to the new place in this PR soon.

**Comment by ilyachur** [comment]
Please let us know, when PR will be ready for review.

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by yangwang201911** [comment]
> Please let us know, when PR will be ready for review.

PR is ready for review now.

File: cmake/features.cmake [hunk 1]
```diff
@@ -107,7 +107,7 @@ ie_option (ENABLE_IR_V7_READER "Enables IR v7 reader" ${ENABLE_IR_V7_READER_DEFA
 
 ie_option (ENABLE_GAPI_PREPROCESSING "Enables G-API preprocessing" ON)
 
-ie_option (ENABLE_MULTI "Enables Multi Device Plugin" ON)
+ie_option (ENABLE_AUTO "Enables Multi Device Plugin" ON)
```
Comment (wangleis) [inline cmake/features.cmake:110]: should be ie_option (ENABLE_AUTO "Enables **_AUTO_** Device Plugin" ON) ?

Comment (yangwang201911) [reply cmake/features.cmake]: Updated.

**Review by wangleis — APPROVED** [review]

File: cmake/features.cmake (hunk 1 above)
Comment (ilya-lavrenov) [reply cmake/features.cmake]: plugin still names as `MULTI`, why the option is renamed?

**Review by myshevts — APPROVED** [review]

File: docs/IE_DG/Deep_Learning_Inference_Engine_DevGuide.md [hunk 2]
```diff
@@ -71,7 +71,7 @@ The table below shows the plugin libraries and additional dependencies for Linux
...
 | HETERO | `libov_hetero_plugin.so`        | Same as for selected plugins                                | `ov_hetero_plugin.dll`       | Same as for selected plugins                                                                           | `libov_hetero_plugin.so`      |  Same as for selected plugins               |
-| MULTI  | `libMultiDevicePlugin.so`   | Same as for selected plugins                                | `MultiDevicePlugin.dll`  | Same as for selected plugins                                                                           | `libMultiDevicePlugin.so` |  Same as for selected plugins               |
+| MULTI  | `libov_auto_plugin.so`   | Same as for selected plugins                                | `ov_auto_plugin.dll`  | Same as for selected plugins                                                                           | `libov_auto_plugin.so` |  Same as for selected plugins               |
```
Comment (ilya-lavrenov) [inline docs/IE_DG/Deep_Learning_Inference_Engine_DevGuide.md:74]: can we add `AUTO` here as well?

File: docs/IE_DG/inference_engine_intro.md [hunk 3]
```diff
@@ -76,7 +76,7 @@ The table below shows the plugin libraries and additional dependencies for Linux
...
 | HETERO | `libov_hetero_plugin.so`        | Same as for selected plugins                                | `ov_hetero_plugin.dll`       | Same as for selected plugins                                                                           | `libov_hetero_plugin.so`      |  Same as for selected plugins               |
-| MULTI  | `libMultiDevicePlugin.so`   | Same as for selected plugins                                | `MultiDevicePlugin.dll`  | Same as for selected plugins                                                                           | `libMultiDevicePlugin.so` |  Same as for selected plugins               |
+| MULTI  | `libov_auto_plugin.so`   | Same as for selected plugins                                | `ov_auto_plugin.dll`  | Same as for selected plugins                                                                           | `libov_auto_plugin.so` |  Same as for selected plugins               |
```
Comment (ilya-lavrenov) [inline docs/IE_DG/inference_engine_intro.md:79]: add `AUTO` as well

File: inference-engine/tests/unit/CMakeLists.txt [hunk 4]
```diff
@@ -28,4 +28,6 @@ if(NGRAPH_ONNX_FRONTEND_ENABLE)
     add_subdirectory(frontends/onnx_import)
 endif()
 
-add_subdirectory(multi)
+if (ENABLE_AUTO)
+add_subdirectory(auto)
```
Comment (ilya-lavrenov) [inline inference-engine/tests/unit/CMakeLists.txt:32]: indentation

File: docs/IE_DG/Deep_Learning_Inference_Engine_DevGuide.md (hunk 2 above)
Comment (yangwang201911) [reply docs/IE_DG/Deep_Learning_Inference_Engine_DevGuide.md:74]: Updated.

File: docs/IE_DG/inference_engine_intro.md (hunk 3 above)
Comment (yangwang201911) [reply docs/IE_DG/inference_engine_intro.md:79]: updated.

File: inference-engine/tests/unit/CMakeLists.txt (hunk 4 above)
Comment (yangwang201911) [reply inference-engine/tests/unit/CMakeLists.txt]: Updated.

File: cmake/features.cmake (hunk 1 above)
Comment (wangleis) [reply cmake/features.cmake]: @ilya-lavrenov CCG prefer AUTO as name of this combined plugin for AUTO and MULTI

Comment (yangwang201911) [reply cmake/features.cmake]: After sync with @myshevts , we keep both the ENABLE_MULTI option and the ENABLE_AUTO option. Auto plugin will be available if any of those two options is enable.

File: cmake/features.cmake [hunk 5]
```diff
@@ -112,7 +112,8 @@ ie_option (ENABLE_IR_V7_READER "Enables IR v7 reader" ${ENABLE_IR_V7_READER_DEFA
 
 ie_option (ENABLE_GAPI_PREPROCESSING "Enables G-API preprocessing" ON)
 
-ie_option (ENABLE_MULTI "Enables Multi Device Plugin" ON)
+ie_option (ENABLE_MULTI "Enables MULTI Device Plugin" ON)
+ie_option (ENABLE_AUTO "Enables AUTO Device Plugin" ON)
```
Comment (ilya-lavrenov) [inline cmake/features.cmake:116]: why do we need 2 options? AUTO == MULTI

Comment (ilyachur) [reply cmake/features.cmake:116]: I propose to have one optionwhich will be aligned with the plugin name

Comment (wangleis) [reply cmake/features.cmake:116]: @ilya-lavrenov @ilyachur 
Just for clarification, AUTO device and MULTI device are different application interface while they are supported by one plugin so file. 
End user may select one of  AUTO device and MULTI device. Then we expect OV still build plugin so file and only expose one device name in plugins.xml.

**Pull request merged**